src/Microsoft.DotNet.Cli.Utils/DotnetRuntimeIdentifiers.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using Microsoft.DotNet.InternalAbstractions;
+
+namespace Microsoft.DotNet.Cli.Utils
+{
+    internal static class DotnetRuntimeIdentifiers
+    {
+        public static IEnumerable<string> InferCurrentRuntimeIdentifiers()
+        {
+            // On non-Windows machines, the CLI may be used on a newer version of a supported OS
+            // and the current RID may not be available in the runtime.* NuGet packages, yet.
+            // so fallback to the RID that was used to build the CLI - which will have the correct
+            // runtime.* NuGet packages available.
+            IEnumerable<string> fallbackIdentifiers = null;
+            if (RuntimeEnvironment.OperatingSystemPlatform != Platform.Windows)
```

Comment:
Because we build the CLI for Windows differently than we build for non-Windows.

When we build for Windows, we build on Win10 machines, and we run those bits on Win7 machines.  So if we used the "BuildRid" (win10-x64) on a Win7 machine, it would be wrong.

But for non-Windows, we are doing the opposite.  We are building on a lower OS version (osx.10.11) and wanting to use those bits on higher OS versions (osx.10.12).

Also, on Windows we are already hard-coding the fallback graph into the CLI code:  https://github.com/dotnet/cli/pull/4281/files#diff-883794133910af7c67b407fd92d10d9dR41, so it isn't necessary to add the "BuildRid" into this graph.

I'll add to the comments here to explain why we don't use BuildRid on Windows.